tokio/src/task/yield_now.rs

```diff
@@ -3,7 +3,17 @@ use std::pin::Pin;
 use std::task::{Context, Poll};
 
 doc_rt_core! {
-    /// Yield execution back to the Tokio runtime.
+    /// Return a `Future` that can be `await`-ed to yield execution back to the
+    /// Tokio runtime.
+    ///
+    /// A task yields by awaiting the returned `Future`, and may resume when
```

Comment:
I'm not sure whether or not we actually want to talk about async functions returning futures, which I think could be confusing given that this function has "no return value". Maybe "awaiting the call to this function"? 

On the other hand, it _does_ return a future...so I am not sure which is less confusing.

Reply:
I guess I tend to be pedantic in rustdoc, but having missed that async keyword the first time, I think its better to highlight it in English.

Reply:
Fair enough!